meson.build

```diff
@@ -22,8 +22,9 @@ dir_include	= 'include'
 cc = meson.get_compiler('c')
 
 cflags = [
+	'-Wall',
+	'-Wextra',
```

Comment:
```
meson.build:49: WARNING: Consider using the built-in warning_level option instead of using "-Wall".
meson.build:49: WARNING: Consider using the built-in warning_level option instead of using "-Wextra".
```

These flags are already added, though, by `default_options: ['warning_level=2']`.

Reply:
Ironically, despite this PR being about fixing build warnings, `CFLAGS=-Werror` is used but not `--fatal-meson-warnings`, so new build warnings cropped up. :D

Reply:
/me mumbles something about compiler warnings only :)

fwiw, I merged this so the cflags entries are the same for meson vs autotools and we don't have to expend braincells on where they're different. `-Wall -Wextra` are a special case, i'm hoping we can get rid of that warning together with autotools at some point.

Reply:
You could include them in meson.build but commented out with a note: "these are handled as a meson builtin option"... :)